### DISCUSSION TITLE
Setting $PATH and Configuring Default AI Vendor/Patterns

### DISCUSSION THREAD

**Discussion by Mualjo** [danielmiessler/Fabric#1232] [Q&A]
Hello Everyone,

I am having a bit of a challenge when it comes to setting my **PATH** right for the command '_fabric_' to work. I followed the instructions from the installation guide. After I reboot my terminal and computer I am having to do the exports for the environment variables all over again for the command. what do I have to do differently to make sure I don't have to keep repeating this process?

When I do have everything operational, I am looking to configure the ### Default AI Vendor and Model and Patterns. I am using Gemini as my AI Vendor, this is the list it gives me for the options i'm going to us for the tool:

Gemini

	[1]	chat-bison-001
	[2]	text-bison-001
	[3]	embedding-gecko-001
	[4]	gemini-1.0-pro-latest
	[5]	gemini-1.0-pro
	[6]	gemini-pro
	[7]	gemini-1.0-pro-001
	[8]	gemini-1.0-pro-vision-latest
	[9]	gemini-pro-vision
	[10]	gemini-1.5-pro-latest
	[11]	gemini-1.5-pro-001
	[12]	gemini-1.5-pro-002
	[13]	gemini-1.5-pro
	[14]	gemini-1.5-pro-exp-0801
	[15]	gemini-1.5-pro-exp-0827
	[16]	gemini-1.5-flash-latest
	[17]	gemini-1.5-flash-001
	[18]	gemini-1.5-flash-001-tuning
	[19]	gemini-1.5-flash
	[20]	gemini-1.5-flash-exp-0827
	[21]	gemini-1.5-flash-002
	[22]	gemini-1.5-flash-8b
	[23]	gemini-1.5-flash-8b-001
	[24]	gemini-1.5-flash-8b-latest
	[25]	gemini-1.5-flash-8b-exp-0827
	[26]	gemini-1.5-flash-8b-exp-0924
	[27]	gemini-2.0-flash-exp
	[28]	gemini-exp-1206
	[29]	gemini-exp-1121
	[30]	gemini-exp-1114
	[31]	gemini-2.0-flash-thinking-exp
	[32]	gemini-2.0-flash-thinking-exp-1219
	[33]	learnlm-1.5-pro-experimental
	[34]	embedding-001
	[35]	text-embedding-004
	[36]	aqa

but which option am I using for the Vendor and Model?

**Comment by NuCl34R** [comment]
Not sure what OS you are using but on a Linux system you have to add the environment variables in your ~/.bashrc file (home directory, make sure show hidden files is checked, or just 'nano .bashrc' when you open the terminal. Save the file, then type 'source .bashrc' to update with the changes, no need to reboot.

As for your 2nd question, you just enter the number next to the Model to use, since the newest version of Gemini is experimental (2.0 flash) then you can use that free of charge for some time, so use option 27, just type 27 when asked for model.

**Comment by bledburn** [comment]
> Hello Everyone,
> 
> I am having a bit of a challenge when it comes to setting my **PATH** right for the command '_fabric_' to work. I followed the instructions from the installation guide. After I reboot my terminal and computer I am having to do the exports for the environment variables all over again for the command. what do I have to do differently to make sure I don't have to keep repeating this process?
> 
> When I do have everything operational, I am looking to configure the ### Default AI Vendor and Model and Patterns. I am using Gemini as my AI Vendor, this is the list it gives me for the options i'm going to us for the tool:
> 
> Gemini
> 
> ```
> [1]	chat-bison-001
> [2]	text-bison-001
> [3]	embedding-gecko-001
> [4]	gemini-1.0-pro-latest
> [5]	gemini-1.0-pro
> [6]	gemini-pro
> [7]	gemini-1.0-pro-001
> [8]	gemini-1.0-pro-vision-latest
> [9]	gemini-pro-vision
> [10]	gemini-1.5-pro-latest
> [11]	gemini-1.5-pro-001
> [12]	gemini-1.5-pro-002
> [13]	gemini-1.5-pro
> [14]	gemini-1.5-pro-exp-0801
> [15]	gemini-1.5-pro-exp-0827
> [16]	gemini-1.5-flash-latest
> [17]	gemini-1.5-flash-001
> [18]	gemini-1.5-flash-001-tuning
> [19]	gemini-1.5-flash
> [20]	gemini-1.5-flash-exp-0827
> [21]	gemini-1.5-flash-002
> [22]	gemini-1.5-flash-8b
> [23]	gemini-1.5-flash-8b-001
> [24]	gemini-1.5-flash-8b-latest
> [25]	gemini-1.5-flash-8b-exp-0827
> [26]	gemini-1.5-flash-8b-exp-0924
> [27]	gemini-2.0-flash-exp
> [28]	gemini-exp-1206
> [29]	gemini-exp-1121
> [30]	gemini-exp-1114
> [31]	gemini-2.0-flash-thinking-exp
> [32]	gemini-2.0-flash-thinking-exp-1219
> [33]	learnlm-1.5-pro-experimental
> [34]	embedding-001
> [35]	text-embedding-004
> [36]	aqa
> ```
> 
> but which option am I using for the Vendor and Model?

NuCl34R has you covered I think. More specifically, understanding your PATH can be frustrating until you understand what it all means. Your system's PATH is a list of directories that are searched for "binaries and executables" so as to make them usable globally. Globally means that they can be called directly from anywhere you happen to be in your file system, within your terminal session. This also means you can call the tool without prefixing it with `./` 

For an in depth walk through, watch here: https://youtu.be/hnFJhubYS88

This will be of particular use to you, if you happen to be on Ubuntu. But if not, rest assured that it will help you to understand your own system and how it operates with bin directories.
